File: 1-Two-Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        int diff, i;

        for (i = 0; i < nums.size(); i++) {
            diff = target - nums[i];
            if (map.count(diff))
                return {map[diff], i};
            map[nums[i]] = i;
        }

        return {-1};
    }
};

File: 1-Two-Sum.ts
function twoSum(nums: number[], target: number): number[] {
  const m = {}

  for (let i = 0; i < nums.length; i++) {
    const diff = target - nums[i]

    if (diff in m) return [i, m[diff]]
    m[nums[i]] = i
  }
  return [-1]
};
